File: Python/assignments/misc/BankAccount.py
class BankAccount:
    bank_name = "First National Dojo"
    all_accounts = []
    def __init__(self, name,int_rate,balance):
        self.int_rate = int_rate
        self.balance = balance
        self.name = name
        BankAccount.all_accounts.append(self)
    @classmethod
    def change_bank_name(cls,name):
        cls.bank_name = name
    @classmethod
    def all_balances(cls):
        sum = 0
        # we use cls to refer to the class
        for account in cls.all_accounts:
            sum += account.balance
        return sum
    @staticmethod
    def can_withdraw(balance,amount):
        if (balance - amount) < 5:
            return False
        else:
            return True
        
    def with_draw(self,amount):
        if BankAccount.can_withdraw(self.balance,amount):
            self.balance -= amount
        else:
            print("Insufficient Funds")
        return self  
    def deposit(self,amount):
        self.balance += amount
        return self
    def yeild_intrest(self):
        self.balance = self.balance * (self.int_rate +1)
        return self
    def display_account_info(self):
        print(self.name,self.int_rate,self.balance)



Zack = BankAccount("Zack", .1, 3000 )
Zack.deposit(10000).deposit(2000).deposit(3000).with_draw(2000).yeild_intrest().display_account_info()


Miles = BankAccount("Miles", .01, 3000 )
Miles.deposit(3000).deposit(100000).with_draw(900).with_draw(100).with_draw(2000).yeild_intrest().display_account_info()

File: Java/bankaccount/bankAccount.java
package bankaccount;

public class bankAccount {

    private double checkingbalance = 0;
    private double savingbalance = 0;
    public static int accounts = 0;
    public static double total = 0;
    
    public bankAccount(){
        bankAccount.accounts +=1; 
        this.checkingbalance = 0;
        this.savingbalance = 0;
    }

    public double getCheckingBalance(){
        return this.checkingbalance;
    }

    public double getSavingBalance(){
        return this.savingbalance;
    }

    public void deposit(double amount, String account){
        if(account == "checking"){
            this.checkingbalance = this.checkingbalance + amount;
            bankAccount.total = total + this.checkingbalance;
        }

        if(account == "saving"){
            this.savingbalance = this.savingbalance + amount;
            bankAccount.total = total + this.savingbalance;
        }
    }

    public void withdraw(double amount, String account){
        if(account == "checking"){
            if(amount - this.checkingbalance != 0){
            this.checkingbalance = this.checkingbalance - amount;
            bankAccount.total = total - this.checkingbalance;
            }
        else{
            System.out.println("Insufficient Funds");
        }
        }

        if(account == "saving"){
            if(amount - this.savingbalance != 0){
            this.savingbalance = this.savingbalance - amount;
            bankAccount.total = total - this.savingbalance;
            }
        else{
            System.out.println("Insufficient Funds");
        }
        }
    }

    public void showBalance(){
        System.out.println("Checking Account Balance: $" + this.checkingbalance);
        System.out.println("Saving Account Balance: $" + this.savingbalance);
        System.out.println("Total Account Balance: $" + bankAccount.total);
    }
    }



